slave/buildslave/monkeypatches/bug4881.py

```diff
@@ -189,7 +191,7 @@ def _resourceFDImplementation(self):
         # to close
         if maxfds > 1024:
             maxfds = 1024
-        return xrange(maxfds)
+        return list(range(maxfds))
```

Comment:
looks overkill I would prefer to make an utility function which does the right thing according to the python version.

You are also changing the functionality. xrange returns an iterator in 2.7, range returns a list in 2.7 and an iterator in 3.0